framework/src/controls/WebServerControl.C

```diff
@@ -0,0 +1,327 @@
+//* This file is part of the MOOSE framework
+//* https://www.mooseframework.org
+//*
+//* All rights reserved, see COPYRIGHT for full restrictions
+//* https://github.com/idaholab/moose/blob/master/COPYRIGHT
+//*
+//* Licensed under LGPL 2.1, please see LICENSE for details
+//* https://www.gnu.org/licenses/lgpl-2.1.html
+
+#include "WebServerControl.h"
+
+#include "minijson.h"
+
+registerMooseObject("MooseApp", WebServerControl);
+
+#define registerWebServerControlCombine1(X, Y) X##Y
+#define registerWebServerControlCombine(X, Y) registerWebServerControlCombine1(X, Y)
+#define registerWebServerControlScalar(T, json_type)                                               \
+  static char registerWebServerControlCombine(wsc_scalar, __COUNTER__) =                           \
+      WebServerControl::registerScalarType<T, json_type>(#T)
+#define registerWebServerControlVector(T, json_type)                                               \
+  static char registerWebServerControlCombine(wsc_vector, __COUNTER__) =                           \
+      WebServerControl::registerVectorType<T, json_type>(#T)
+#define registerWebServerControlScalarBool(T)                                                      \
+  registerWebServerControlScalar(T, miniJson::JsonType::kBool)
+#define registerWebServerControlScalarNumber(T)                                                    \
+  registerWebServerControlScalar(T, miniJson::JsonType::kNumber)
+#define registerWebServerControlScalarString(T)                                                    \
+  registerWebServerControlScalar(T, miniJson::JsonType::kString)
+#define registerWebServerControlVectorNumber(T)                                                    \
+  registerWebServerControlVector(T, miniJson::JsonType::kNumber)
+#define registerWebServerControlVectorString(T)                                                    \
+  registerWebServerControlVector(T, miniJson::JsonType::kString)
+
+// Registration of the types that we can accept in the web server for controlling parameters
+registerWebServerControlScalarBool(bool);
+registerWebServerControlScalarNumber(Real);
+registerWebServerControlScalarString(std::string);
+registerWebServerControlVectorNumber(Real);
+registerWebServerControlVectorString(std::string);
```

Comment:
why those macros for exactly one single use? are you foreseeing `registerWebServerControlScalarNumber(int);` etc. ?

Reply:
I likely already have a need for `ScalarNumber(unsigned int)` and `VectorNumber(unsigned int)`

I did my best to just generalize this as a way to take json fields and convert them into c++ types